### PR TITLE
[QI2-793] Qiskit backend support

### PR DESCRIPTION
**Pull request by napocornejo**
Integrate qiskit

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -28,6 +28,10 @@ pydantic = "^2.6.3"
 qi-compute-api-client = {git = "https://github.com/QuTech-Delft/compute-api-client.git", branch="0.26.0"}
 qxelarator = {version = "^0.6.2", optional = true}
 pydantic-settings = "^2.2.1"
+qiskit = "^0.45.2"
+qiskit-algorithms = "^0.2.2"
+qiskit-nature = "^0.7.1"
+pyscf = {version = "^2.4.0", markers = 'sys_platform != "win32"'}
 
 [tool.poetry.extras]
 local = ["qxelarator"]
```
Comment (koffie) [inline pyproject.toml:37]: Make it work with:
```
[tool.coverage.run]
branch = true
source = ["quantuminspire"]
omit = ["tests/*", "**/.tox/**"]
```
and also copy over the tests.

Comment (napocornejo) [reply pyproject.toml:37]: Done

File: quantuminspire/sdk/qiskit/backend.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    def run(  # pylint: disable=too-many-locals
+        self,
+        run_input: Union[QuantumCircuit, List[QuantumCircuit]],
+        shots: Optional[int] = None,
+        memory: Optional[bool] = None,
+        allow_fsp: bool = True,
+        **run_config: Dict[str, Any],
+    ) -> QiskitQuantumInspireJob:
+
+        del allow_fsp
```
Comment (koffie) [inline quantuminspire/sdk/qiskit/backend.py]: Could we actually make sure it is passed down to the CirquitToString parser

File: tests/sdk/qiskit/test_backend.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        assert status.pending_jobs == 0
+        assert status.operational == True
+        assert status.status_msg == "online"
+
+    def test_run_normal(self, backend):
+        qc = QuantumCircuit(2, 2)
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.measure([0, 1], [0, 1])
+
```
Comment (koffie) [inline tests/sdk/qiskit/test_backend.py:79]: Could you add a test that shows the explicitly generated cqasm string?

Comment (napocornejo) [reply tests/sdk/qiskit/test_backend.py:79]: There is now, in _TestQiskitToCQasm::test_generate_cqasm_with_bell_circuit_

File: quantuminspire/sdk/qiskit/backend.py (hunk 2 above)
Comment (napocornejo) [reply quantuminspire/sdk/qiskit/backend.py]: Done.

**Review by koffie — APPROVED** [review]

**Pull request merged**